### PR TITLE
Feature/Add engagements API

### PR DESCRIPTION
**Pull request by eikooc**
added support for Engagements api (Get, Create, Update, Delete and ListAssociated)

### CODE REVIEW

File: Vertical.HubSpot.Api/Engagements/EngagementsApi.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        public async Task<T> GetEngagement<T>(long id)
+            where T : HubSpotEngagementResult
+        {
+            JObject response = await rest.Get<JObject>($"/engagements/v1/engagements/{id}");
+            return ToEngagementResult<T>(response);
+        }
+
+        public async Task<JObject> DeleteEngagement(long id)
+        {
+            JObject response = await rest.Delete<JObject>($"/engagements/v1/engagements/{id}");
```
Comment (telmengedar) [inline Vertical.HubSpot.Api/Engagements/EngagementsApi.cs]: if you return the rest response directly here anyways you could drop the await/async statement and save a task wrap operation.

Comment (eikooc) [reply Vertical.HubSpot.Api/Engagements/EngagementsApi.cs]: Alright

File: Vertical.HubSpot.Api/Engagements/HubspotEngagement.cs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        [JsonProperty("taskType")]
+        public string TaskType { get; set; }
+
+        [JsonProperty("timestamp")]
+        public long Timestamp { get; set; }
+
+        [JsonProperty("uuid")]
+        public string Uuid { get; set; }
+    }
+}
```
Comment (telmengedar) [inline Vertical.HubSpot.Api/Engagements/HubspotEngagement.cs:62]: multiple classes in a single file? ... i would rather see them separated

**Review by eikooc — COMMENTED** [review]
I'll change it immediately

File: Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        Task<T> UpdateEngagement<T>(long id, HubSpotEngagementResult data)
+            where T : HubSpotEngagementResult;
+
+        /// <summary>
+        /// creates or updates an engagement
+        /// </summary>
+        /// <typeparam name="T">type of engagement model</typeparam>
+        /// <param name="contactId">id of contact to create or update</param>
+        /// <returns>A list of engagements associated with the user</returns>
+        Task<PageResponse<T>> ListAssociatedEngagements<T>(ObjectType type, long contactId, long? offset = null, int? count = null)
```
Comment (telmengedar) [inline Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs:29]: only one of the parameters is included in xml-docs

File: Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+using Vertical.HubSpot.Api.Data;
+
+namespace Vertical.HubSpot.Api.Engagements
+{
+    public enum ObjectType
+    {
+        Contact, 
+        Company,
+        Deal
+    }
```
Comment (telmengedar) [inline Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs]: this should get into its own file, makes it awkward to find otherwise

**Review by telmengedar — CHANGES_REQUESTED** [review]
A few types are merged into one file. Usually i like one type/class/struct per file

And there are quite a few xml-docs missing. In the implementation you could use **inheritdoc/** to not clone the xml-doc of the interface, but also enums and types.
If it was only a few missing comments i might not be as strict but sadly this PR is missing a lot

Other than that it seems to be functional by what i can see, so if these style issues are fixed it can get merged.

File: Vertical.HubSpot.Api/Engagements/HubspotEngagement.cs (hunk 2 above)
Comment (eikooc) [reply Vertical.HubSpot.Api/Engagements/HubspotEngagement.cs:62]: They are separated now.

File: Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs (hunk 3 above)
Comment (eikooc) [reply Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs:29]: Both are included now.

**Review by eikooc — COMMENTED** [review]
The requested changes are now submitted

File: Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs (hunk 4 above)
Comment (eikooc) [reply Vertical.HubSpot.Api/Engagements/IEngagementsApi.cs]: Moved to it's own file

**Review by telmengedar — APPROVED** [review]
*sigh* ... quite a few xml-docs are still missing but i don't feel like blocking today :P
but keep in mind that xml-docs make packages a lot more "pleasant" to use (because intellisense and such)

**Comment by eikooc** [comment]
Alright. I'll add them in the coming days. Maybe we should add a README to.

**Pull request merged**